src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -179,6 +180,11 @@ public void RegisterWorkspaceHost(IVisualStudioWorkspaceHost host)
             _workspaceHosts.Add(new WorkspaceHostState(this, host));
         }
 
+        public Task RegisterWorkspaceHostFromForegroundThreadAsync(IVisualStudioWorkspaceHost host)
```

Comment:
oh. well, don't want to do big change for this. so where caller get foreground thread scheduler?